src/datasets.rs

```diff
@@ -0,0 +1,390 @@
+    pub annotators_can_create_tags: Option<bool>,
+    pub annotators_can_instantiate_workflows: Option<bool>,
+    pub anyone_can_double_assign: Option<bool>,
+
+    // TODO: annotations
+    #[serde(skip)]
+    pub annotation_classes: Vec<String>,
+
+    pub default_workflow_template_id: Option<u32>,
+
+    pub id: u32,
+    pub inserted_at: Option<String>,
+    pub instructions: Option<String>,
+
+    pub name: Option<String>,
+    pub num_annotations: Option<Option<u32>>,
+    pub num_annotators: Option<Option<u32>>,
+    pub num_classes: Option<u32>,
+    pub num_complete_files: Option<u32>,
+    pub num_images: Option<u32>,
+    pub num_items: Option<u32>,
+    pub num_videos: Option<u32>,
+    pub owner_id: Option<u32>,
+    pub parent_id: Option<u32>,
+    pub pdf_fit_page: Option<bool>,
+    pub progress: Option<f64>,
+    pub public: Option<bool>,
+    pub reviewers_can_annotate: Option<bool>,
+    pub slug: String,
+    pub team_id: Option<u32>,
+    pub team_slug: Option<String>,
+
+    // TODO: thumbnails
+    #[serde(skip)]
+    pub thumbnails: Option<Vec<String>>,
+    pub updated_at: Option<String>,
+    pub version: Option<u32>,
+    pub work_size: Option<u32>,
+    pub work_prioritization: Option<String>,
+}
+
+// TODO: Fix this when working on the annotation aspects of the APi
+#[derive(Debug, Default, Clone, Serialize, Deserialize, Dummy, PartialEq)]
+pub struct AnnotationClass {
+    pub id: u32,
+    pub name: String,
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize, Dummy, PartialEq)]
+pub struct AnnotationType {
+    pub id: u32,
+    pub count: u32,
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize, Dummy, PartialEq)]
+pub struct ExportMetadata {
+    pub annotation_classes: Vec<AnnotationClass>,
+    pub annotation_types: Vec<AnnotationType>,
+}
+#[derive(Debug, Default, Clone, Serialize, Deserialize, Dummy, PartialEq)]
+pub struct Export {
+    pub download_url: String,
+    pub format: String,
+    pub inserted_at: String,
+    pub latest: bool,
+    pub metadata: ExportMetadata,
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize, Dummy, PartialEq)]
+struct DatasetName {
+    pub name: String,
+}
+
+impl Dataset {
+    pub async fn assign_items(
+        &self,
+        client: &V7Client,
+        assignee_id: &u32,
+        filter: &Filter,
+    ) -> Result<()> {
+        #[derive(Serialize, Deserialize)]
+        struct Payload {
+            pub assignee_id: u32,
+            pub filter: Filter,
+        }
+
+        let payload = Payload {
+            assignee_id: *assignee_id,
+            filter: filter.clone(),
+        };
+
+        let response = client
+            .post(&format!("datasets/{}/assign_items", self.id), &payload)
+            .await?;
+
+        let status = response.status();
+
+        // 204 is correct operation for this endpoint
+        if status != 204 {
+            bail!("Invalid status code {status}")
```

Comment:
**thought (non-blocking):** I wonder if we could make some custom errors to return here instead of `bail!`'ing in the future, to return some more specific errors to clients that use it 🤔

Reply:
💯  I've raised this issue to track this work #5